src/environment.jl

```diff
@@ -341,3 +341,28 @@ function has_cuda()
         return parse(Bool, flag)
     end
 end
+
+"""
+    MPI.has_rocm()
+
+Check if the MPI implementation is known to have ROCm support. Currently only Open MPI
+provides a mechanism to check, so it will return `false` with other implementations
+(unless overriden).
+
+This can be overriden by setting the `JULIA_MPI_HAS_ROCM` environment variable to `true`
+or `false`.
+"""
+function has_rocm()
+    flag = get(ENV, "JULIA_MPI_HAS_ROCM", nothing)
+    if flag === nothing
+        # Only Open MPI provides a function to check ROCm support
+        @static if MPI_LIBRARY == "OpenMPI"
```

Comment:
We probably need to use `MPI_LIBRARY_VERSION` to check that this is indeed `v5`

Reply:
The current compat is on 4. Should we hold it off till newer JLLs are supported?

Reply:
Someone will have to debug what's the problem with OpenMPI 5: https://github.com/JuliaParallel/MPI.jl/pull/789#discussion_r1382022041

Reply:
Someone could already run with a local version of OpenMPI 5 before we get around to jll

Reply:
If that helps for testing, here's a docker container with openmpi v5.0.2: https://github.com/JuliaParallel/github-actions-buildcache/pkgs/container/github-actions-buildcache/183303325?tag=openmpi-5.0.2-wufrzwmzj3h642rou7vsrphzpysokjui.spack

Reply:
For the record, `MPI.jl` tests pass in that container, I'll open a PR to add that to the matrix so that we test OpenMPI v5.  We still need to figure out what was the problem with the JLL.

Reply:
@avik-pal with #826 now merged we actually have coverage for OpenMPI v5 on CI (although not with an AMD GPU available...) so you can do a test like
```suggestion
        @static if MPI_LIBRARY == "OpenMPI" && MPI_LIBRARY_VERSION >= v"5"
```
as suggested above by @vchuravy and we can at least ensure the ccall is working correctly.

Reply:
Tested locally that ccall works by bumping the jll compat to 5

```julia
julia> using MPI

julia> MPI.has_cuda()
false

julia> MPI.has_rocm()
false
```